### PR TITLE
ENH: stats.spearmanr: add statistic attribute to result object as ali…

### PR DESCRIPTION
**Pull request by JozsefKutas**
#### Reference issue
https://github.com/scipy/scipy/issues/16364

#### What does this implement/fix?
Switch to using `SignificanceResult` as a result object for `spearmanr`, and add `correlation` attribute as an alias for `statistic` for backward compatibility.

### CODE REVIEW

File: scipy/stats/_mstats_basic.py [hunk 1]
```diff
@@ -591,10 +588,19 @@ def spearmanr(x, y=None, use_ties=True, axis=None, nan_policy='propagate',
...
+        statistic : float or ndarray (2-D square)
+            Spearman correlation matrix or correlation coefficient (if only 2
+            variables are given as parameters). Correlation matrix is square
+            with length equal to total number of variables (columns or rows) in
+            ``a`` and ``b`` combined.
+        pvalue : float
+            The p-value for a hypothesis test whose null hypothesis
+            is that two sets of data are uncorrelated. See `alternative` above
+            for alternative hypotheses. `pvalue` has the same
+            shape as `statistic`.
```
Comment (mdhaber) [inline scipy/stats/_mstats_basic.py:603]:
```suggestion
            is that two sets of data are linearly uncorrelated. See
            `alternative` above for alternative hypotheses. `pvalue` has the
            same shape as `statistic`.
```

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -4794,16 +4791,19 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
...
+        statistic : float or ndarray (2-D square)
+            Spearman correlation matrix or correlation coefficient (if only 2
+            variables are given as parameters). Correlation matrix is square
+            with length equal to total number of variables (columns or rows) in
+            ``a`` and ``b`` combined.
+        pvalue : float
+            The p-value for a hypothesis test whose null hypothesis
+            is that two sets of data are uncorrelated. See `alternative` above
+            for alternative hypotheses. `pvalue` has the same
+            shape as `statistic`.
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:4806]:
```suggestion
            is that two sets of data are linearly uncorrelated. See
            `alternative` above for alternative hypotheses. `pvalue` has the
            same shape as `statistic`.
```

File: scipy/stats/_stats_py.py [hunk 3]
```diff
@@ -4822,41 +4822,49 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
     --------
     >>> import numpy as np
     >>> from scipy import stats
-    >>> stats.spearmanr([1,2,3,4,5], [5,6,7,8,7])
-    SpearmanrResult(correlation=0.82078..., pvalue=0.08858...)
+    >>> res = stats.spearmanr([1,2,3,4,5], [5,6,7,8,7])
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:4825]:
```suggestion
    >>> res = stats.spearmanr([1, 2, 3, 4, 5], [5, 6, 7, 8, 7])
```

File: scipy/stats/_stats_py.py [hunk 4]
```diff
@@ -4822,41 +4822,49 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
...
-    >>> stats.spearmanr(x2n)
-    SpearmanrResult(correlation=-0.07960396039603959, pvalue=0.4311168705769747)
-    >>> stats.spearmanr(x2n[:,0], x2n[:,1])
-    SpearmanrResult(correlation=-0.07960396039603959, pvalue=0.4311168705769747)
-    >>> rho, pval = stats.spearmanr(x2n, y2n)
-    >>> rho
+    >>> res = stats.spearmanr(x2n)
+    >>> res.statistic, res.pvalue
+    (-0.07960396039603959, 0.4311168705769747)
+    >>> res = stats.spearmanr(x2n[:,0], x2n[:,1])
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:4836]:
```suggestion
    >>> res = stats.spearmanr(x2n[:, 0], x2n[:, 1])
```

**Review by mdhaber — APPROVED** [review]
With these changes, LGTM.

**Review by tupui — APPROVED** [review]
LGTM, thanks @JozsefKutas

File: scipy/stats/_mstats_basic.py [hunk 5]
```diff
@@ -591,10 +588,19 @@ def spearmanr(x, y=None, use_ties=True, axis=None, nan_policy='propagate',
 
     Returns
     -------
-    correlation : float
-        Spearman correlation coefficient
-    pvalue : float
-        2-tailed p-value.
+    res: SignificanceResult
```
Comment (tupui) [inline scipy/stats/_mstats_basic.py:591]:
```suggestion
    res : SignificanceResult
```

File: scipy/stats/_stats_py.py [hunk 6]
```diff
@@ -4794,16 +4791,19 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
...
-        Spearman correlation matrix or correlation coefficient (if only 2
-        variables are given as parameters. Correlation matrix is square with
-        length equal to total number of variables (columns or rows) in ``a``
-        and ``b`` combined.
-    pvalue : float
-        The p-value for a hypothesis test whose null hypotheisis
-        is that two sets of data are uncorrelated. See `alternative` above
-        for alternative hypotheses. `pvalue` has the same
-        shape as `correlation`.
+    res: SignificanceResult
```
Comment (tupui) [inline scipy/stats/_stats_py.py:4794]:
```suggestion
    res : SignificanceResult
```

**Pull request merged**